sandbox/libs/authn/src/main/java/org/opensearch/authn/PermissionFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is used to create Permission instances. The factory can create both standard Permissions which have specific
+ * formatting requirements and legacy permissions which are not checked for validity on creation.
+ */
+public class PermissionFactory {
+
+    public final static String[] INVALID_CHARACTERS = new String[] { ":", "" }; // This is a placeholder for what may want to be banned
+
+    // A placeholder for the different resources which a permission may grant a permission based on
+    public final static String[] QUALIFIED_PERMISSION_TYPES = new String[] { "cluster", "indices", "plugin", "extension" };
```

Comment:
This looks like a good use for an enum

Reply:
I will swap it!

Reply:
So the issue with the enum use is that there is not a clean solution to checking the membership of the provided string. With the array/array list you can use `.contains()` but with enums the equivalent is manual iteration.

Reply:
You can use [EnumUtils.java](https://javadevcentral.com/enumutils-from-apache-commons-lang) class from apache commons-lang as we already use this library. There is a [better implementation](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/EnumUtils.html) of the same class in commons-lang3 but we do not import it.